pcsx2/gui/MainMenuClicks.cpp

```diff
@@ -902,31 +902,32 @@ void MainEmuFrame::Menu_Recording_Play_Click(wxCommandEvent &event)
 	const bool initiallyPaused = g_InputRecordingControls.IsRecordingPaused();
 	if (!initiallyPaused)
 		g_InputRecordingControls.PauseImmediately();
+
 	wxFileDialog openFileDialog(this, _("Select P2M2 record file."), L"", L"",
 		L"p2m2 file(*.p2m2)|*.p2m2", wxFD_OPEN);
-	if (openFileDialog.ShowModal() == wxID_CANCEL)
+	if (openFileDialog.ShowModal() == wxID_CANCEL && !initiallyPaused)
```

Comment:
If a user manually pauses before opening the file dialog and then cancels, this condition fails. It would then start trying to load a file path that wasn't confirmed and/or doesn't exist. Additionally, if a file was already loaded, that file will now get closed even when the user didn't specifically tell the emulator to do that.
In comparison to what I had before, as long as cancel was chosen, it will _always_ cancel this action regardless of whether emulation was manually paused or not.
I don't see why this was changed.

Reply:
Yeah I see my mistake in both this and below.  The reason this code was adjusted wasn't to fix a bug, but just to try to simplify it.  Lots of nested if statements